### PR TITLE
Added a function to generate report data, from a report model instance

### PR DESCRIPTION
**Pull request by upatricck**
## DONE

* Modified the structure of the model to contain a Report model.
* Added a function to generate report data from a report model instance
* A function to send the reports
* Creating a task to trigger the reports.
* Modifying the email templates
* Adding a ui to add and remove reports
* Enhancing the ui (will be done in when we start enhancing the ui)

### CODE REVIEW

File: seshdash/utils/reporting.py [hunk 1]
```diff
@@ -81,3 +82,31 @@ def prepare_report(site, duration="week"):
...
+    The function receives a report model instance and
+    it returns a dict containing the aggregated value of the 
+    report attributes
+    """ 
+    report_data = {}
+    now = datetime.now()
+    
+    # Getting the correct date range
+    if report.duration == "daily":
+        start = now - relativedelta(hours=24)
```
Comment (jlark) [inline seshdash/utils/reporting.py]: How does this relativedelta work?

File: seshdash/utils/reporting.py [hunk 2]
```diff
@@ -81,3 +82,31 @@ def prepare_report(site, duration="week"):
...
+    elif report.duration == "weekly":
+        start = now - relativedelta(weeks=1)
+    elif report.duration == "monthly":
+        start = now - relativedelta(months=1)
+
+    # Getting the aggregation of the values in the report attributes
+    for attribute in report.attributes:
+        table = apps.get_model(app_label="seshdash", model_name=attribute["table"])
+        data = table.objects.filter(site=report.site
+                                    date__range=[start, now]).aggregate(Sum(attribute["column"]))
```
Comment (jlark) [inline seshdash/utils/reporting.py]: the summation here is a bit tricky. we might want to do other things besides summations. Perhaps leave a TODO here.

File: seshdash/utils/reporting.py (hunk 1 above)
Comment (upatricck) [inline seshdash/utils/reporting.py]: Relative delta works the same as timedelta except it has many parameters like hours and months.

Comment (upatricck) [inline seshdash/utils/reporting.py]: We replaced the all the migrations with the 0001_inital_.py

Comment (jlark) [inline seshdash/utils/reporting.py]: check out the changed files in the commit there are many old and new migrations.

Comment (upatricck) [inline seshdash/utils/reporting.py]: I see only one new and others removed.

Comment (jlark) [inline seshdash/utils/reporting.py]: sorry yes youre correct

**Comment by jlark** [comment]
Is this ready to go?

**Comment by upatricck** [comment]
Its ready to go.

**Comment by jlark** [comment]
@upatricck see comment on monthly report.

**Pull request merged**